content/articles/get-geolocation-by-ip-address-and-country/index.md

```diff
@@ -39,40 +39,151 @@ Due to a shortage of data collection methods, geolocation is currently in progre
 
-This is the mark of appearance code that allows consumers to be viewed based on their current geolocation data. It also exchanges the district association class and imports the region association class. When the IP address is valid and produces a lot of data, it asks for the geolocation. Alternatively, something terrible may occur, in which case the UI will be informed through the oversight message.
+### Get and validate current IP address
 
-- index.php
-
-  ![layout](layout.jpg)
-
-### Plan a PHP cURL API call for geolocation data.
+This is the home page code which contains the HTML code to acknowledge users with the current geolocation data. It imports the location service class invokes the methods to get and validate the IP address. Once the IP is validated and returns true, it requests the geolocation data. Or else, it will display the error message to the UI.
 
-`$_SERVER` is used to form an if-else conditional ladder for the IP address acquisition.
-`GetLocation()` may utilize cURL to access the `ipwhois API` when an IP address is supported, which gives a large amount of data. Afterward, the API will return a JSON object. Geolocation data is used to determine the nation's location.
+- index.php
 
-![layout-2](layout-2.jpg)
+```php
+<?php
+require_once 'lib/Request.php';
+$requestModel = new Request();
+$ip = $requestModel->getIpAddress();
+$isValidIpAddress = $requestModel->isValidIpAddress($ip);
+?>
+<HTML>
+<HEAD>
+<TITLE>Get Geo Location by the IP address</TITLE>
+<link href="assets/css/style.css" type="text/css" rel="stylesheet" />
+</HEAD>
+<BODY>
+	<div class="txt-heading">Get Geo Location by the IP address</div>
+			<?php
+if ($isValidIpAddress == "") {
+    echo "<div class='error'>Invalid IP address $ip</div>";
+} else {
+    $geoLocationData = $requestModel->getLocation($ip);
+    print "<PRE>";
+    print_r($geoLocationData);
+    ?>
+	<div id="location">
+		<div class="geo-location-detail">
+
+			<div class="row">
+				<div class="form-label">
+					Country Name: <?php  echo $geoLocationData['country'];?>
+				</div>
+			</div>
+			<div class="row">
+				<div class="form-label">
+					Country Code: <?php   echo $geoLocationData['country_code'];?>
+				</div>
+			</div>
+			<div class="row">
+				<div class="form-label">
+					Ip Address: <?php  echo $geoLocationData['ip'];?>
+				</div>
+			</div>
+		</div>
+	</div>
+<?php }?>
+</BODY>
+</HTML>
+```
+
+### Prepare API request to get geolocation via PHP cURL
+
+The `getIPAddress()` function builds an if-else-if ladder of the majority of the scenario to get the non-empty IP address using the `$_SERVER` variable. Once the IP is validated and returns true, the `getLocation()` function to request the ipwhois API via cURL. The API will return a JSON response as a result. This example decodes the JSON response and parses the geolocation data to get the country details from it.
+
+```php

```

Comment:
Could you separate the code based on each function, for a better understanding to the reader?